corehq/motech/generic_inbound/views.py

```diff
@@ -144,15 +146,47 @@ def generic_inbound_api(request, domain, api_id):
         return JsonResponse({'error': str(e)}, status=400)
 
     try:
-        data = api.parsed_expression(context.root_doc, context)
+        response = _execute_case_api(
+            request.domain,
+            request.couch_user,
+            request.META.get('HTTP_USER_AGENT'),
+            context,
+            api
+        )
     except BadSpecError as e:
         return JsonResponse({'error': str(e)}, status=500)
-
-    response = _execute_case_api(data)
+    except UserError as e:
+        return JsonResponse({'error': str(e)}, status=400)
+    except SubmissionError as e:
+        return JsonResponse({
+            'error': str(e),
+            'form_id': e.form_id,
+        }, status=400)
```

Comment:
I was thinking about this recently in the context of the v0.6 case API - should we anticipate catching all these sorts of errors in the API layer?  Like, if we do get a submission error, is that a user error or a server error?  I could see a future world where we have project-specific validation built in to form processing, in which case a 400 would make a lot of sense, but I wonder whether in the current world a 500 would be better.
(not relevant to this PR)

Reply:
Yea, I wasn't sure about that and since this is just the MVP I didn't want to do too much other work. I think we can re-visit when we start having to transform the responses.